components/Map.js

```diff
@@ -13,61 +14,179 @@ const basemap_url =
-        iconUrl: '../static/t.svg',
+        iconUrl: '../static/marker-11-red.svg',
         iconSize: [20, 20],
       }),
       mv: new L.Icon({
-        iconUrl: '../static/b.svg',
+        iconUrl: '../static/marker-11-blue.svg',
         iconSize: [20, 20],
       }),
       bike: new L.Icon({
-        iconUrl: '../static/cone.svg',
+        iconUrl: '../static/marker-11-yellow.svg',
         iconSize: [20, 20],
       }),
     };
 
-    // Set up THE MAP
-    this.map = L.map(el)
-      .setView([42.357004, -71.062309], 12)
-
-      // Add crash data and symbolize by the type of crash
-      .addLayer(
-        featureLayer({
-          url: feature_service_url,
-          pointToLayer: function(geojson, latlng) {
-            return L.marker(latlng, {
-              icon: icons[geojson.properties.mode_type.toLowerCase()],
-            });
-          },
-        })
-      )
-      // Boston basemap only includes Boston, so we layer over Esri's "Gray"
-      // basemap.
+    this.features = featureLayer({
+      url: feature_service_url,
+      pointToLayer: function(geojson, latlng) {
+        return L.marker(latlng, {
+          icon: icons[geojson.properties.mode_type.toLowerCase()],
+        });
+      },
+      onEachFeature: this.bindPopUp,
+    });
+
+    // Set features to load based on the default from and to dates
+    const dateQuery =
+      'dispatch_ts >=' +
+      "'" +
+      this.props.fromDate +
+      "'" +
+      ' AND dispatch_ts <= ' +
+      "'" +
+      this.props.toDate +
+      "'";
+
+    this.features.setWhere(dateQuery);
+
+    // Add basemap layers and features to map
+    this.map
       .addLayer(basemapLayer('Gray'))
-      .addLayer(tiledMapLayer({ url: basemap_url }));
-  };
+      .addLayer(tiledMapLayer({ url: basemap_url }))
+      .addLayer(this.features);
+
+    // Query for feature counts and update this.state.crashCounts
+    this.features
+      .query()
+      .where(dateQuery)
+      .run((error, featureCollection) => {
+        this.setState({
+          crashCounts: featureCollection.features.length.toString(),
+        });
+      });
+
+    // Query for last updated date
+    this.features
+      .query()
+      .where('1=1')
+      .orderBy('dispatch_ts', 'DESC')
+      .run((error, featureCollection) => {
+        const mostRecent = new Date(
+          featureCollection.features[0].properties.dispatch_ts
+        );
+        const month = mostRecent.getMonth();
+        const year = mostRecent.getFullYear();
+        const lastUpdate = month + 1 + '/' + year;
 
-  componentWillReceiveProps() {
-    // Check if mode filter has changed
+        this.setState({ lastUpdatedDate: lastUpdate });
+      });
+  }
+
+  componentWillReceiveProps({ modeSelection, fromDate, toDate }) {
+    if (
+      this.props.modeSelection !== modeSelection ||
+      this.props.fromDate !== fromDate ||
+      this.props.toDate !== toDate
+    ) {
+      if (modeSelection == 'any') {
+        const featuresQuery =
```

Comment:
changed the name here to “loadQuery” to be more specific, left it as defined here, but pulled the query building in componentswillmount out since they ended up having to be build differently than this one